### PR TITLE
Define precisions based on dictionary instead of inferring from API

### PR DESCRIPTION
**Pull request by auterium**
As discussed with @kroitor on #5960, the way that the markets' precision is inferred from the API is unreliable in Bitso. This PR proposes using a dictionary to define the default precision (8) with some overrides for the known symbols that have a smaller precision (XRP 6, MXN 2, tUSD 2).

It's been quite some years since I quit PHP in favor of JS and I have some basic Python knowledge, so please go easy on me with the inclusion of changes on these 2 languages 😄

### CODE REVIEW

**Comment by kroitor** [comment]
Thanks @auterium !

> It's been quite some years since I quit PHP in favor of JS and I have some basic Python knowledge, so please go easy on me with the inclusion of changes on these 2 languages 😄

We are transpiling JS to Python and PHP, so your edits to Python and PHP files will be overwritten anyway, most of our py/php code is generated from js, more about it here: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#multilanguage-support ;)

**Comment by auterium** [comment]
Thanks @kroitor too bad that it will be overriden, but it felt nice to dust off that PHP 😛

File: js/bitso.js [hunk 1]
```diff
@@ -118,8 +126,8 @@ module.exports = class bitso extends Exchange {
                 },
             };
             const precision = {
-                'amount': this.precisionFromString (market['minimum_amount']),
-                'price': this.precisionFromString (market['minimum_price']),
+                'amount': this.options.precision[base] ? this.options.precision[base] : this.options.precision.default,
```
Comment (kroitor) [inline js/bitso.js]: We need to fix dictionary keys to conform to this rule:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes

> all associative keys must be single-quoted strings everywhere, `array['good']`, `array.bad`

So,

```JavaScript
const precision = {
    'amount': this.options.precision[base] ? this.options.precision[base] : this.options.precision.default,
    'price': this.options.precision[quote] ? this.options.precision[quote] : this.options.precision.default,
};
```

↓

```JavaScript
const precision = {
    'amount': this.safeInteger (this.options['precision'], base, this.options['defaultPrecision']),
    'price': this.safeInteger (this.options['precision'], quote, this.options['defaultPrecision']),
};
```

However, this is not the end of the story. We should not mix base precision with quote precision, I believe (otherwise we might get all sorts of unexpected behaviour), so, we have to add subkeys `'base'` and `'quote'` to have `options['precision']['base'][currencyCode]` and `options['precision']['quote'][currencyCode]`, I think, though, I may be wrong on this one. Correct me if that's the case.

In other words, we probably should end up with something like

```JavaScript
const precision = {
    'amount': this.safeInteger (this.options['precision']['base'], base, this.options['defaultPrecision']),
    'price': this.safeInteger (this.options['precision']['quote'], quote, this.options['defaultPrecision']),
};
```

It would be very helpful if you could edit the above to make it transpileable. Let me know if you have any questions, will be happy to assist.

Comment (kroitor) [reply js/bitso.js]: P.S. We also should not mix `defaultPrecision` into the precision options, because there's no currency named `defaultPrecision`, so it's better to have it one level higher, in the options, as a standalone flag.

Comment (auterium) [reply js/bitso.js]: In the case of Bitso, a given symbol/currency will have the same precision everywhere (BTC has 8 a precision of 8 both when it's a quote or a base). I can do a separation of base & quotes if this makes more sense to you

Comment (kroitor) [reply js/bitso.js]: > In the case of Bitso, a given symbol/currency will have the same precision everywhere (BTC has 8 a precision of 8 both when it's a quote or a base). I can do a separation of base & quotes if this makes more sense to you

No, I totally trust you on this, so, if you say we can get away with just one assoc dict – that's perfectly fine. Let's not have that separation then (as is). It would be great if you could fix those minor transpilation errors, though. Let me know if anything. Thx again!

**Comment by kroitor** [comment]
This is a great and solid contribution Marc! Thank you again for your time fixing it! Merging! )
